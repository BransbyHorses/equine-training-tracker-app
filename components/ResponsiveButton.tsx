import { Button, useTheme } from "@mui/material";
import React from "react";

const ResponsiveButton = (props: any) => {
	const theme = useTheme();
	return (
		<Button
			variant="contained"
			sx={{
				backgroundColor: "primary.light",
				[theme.breakpoints.between("xs", "lg")]: {
					width: "90%",
					position: "absolute",
					bottom: "24px",
					left: "50%",
					transform: "translate(-50%, -50%)",
				},
				[theme.breakpoints.between("lg", "xl")]: {
					...props.desktopStyles,
				},
			}}
			{...props}
		>
			{props.children}
		</Button>
	);
};

export default ResponsiveButton;
