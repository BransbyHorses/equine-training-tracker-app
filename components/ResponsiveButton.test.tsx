import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import ResponsiveButton from "./ResponsiveButton";

describe("responsive button", () => {
	it("renders with no props", () => {
		render(<ResponsiveButton>Save</ResponsiveButton>);
		const saveButton = screen.getByRole("button");
        expect(saveButton).toBeInTheDocument();
        expect(saveButton).toMatchSnapshot();
	});
});
