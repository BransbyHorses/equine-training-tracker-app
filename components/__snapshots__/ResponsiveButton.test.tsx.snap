// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsive button renders with no props 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-z6a6p6-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Save
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
